File: 2D/saving_in_neigh.m
for j=1:3
    if(neigh(id_tri,points_together(i))~= -1)
        if(neigh(neigh(id_tri,points_together(i)),j)==id_tri)
            triangle(neigh(id_tri,points_together(i)),j+3)=...
            tr(points_together(i)+3);
            triangle(neigh(id_tri,points_together(i)),j+6) = s_temp(i);
        end
    end
end

File: C/saving_in_neigh.cpp
#include "global_variables.h"
#include "functions.h"

void saving_in_neigh(){
  for( int j = 0; j < 3; j++){
    if(neigh[neigh[id_tri-1][points_together[i]-1][j] == id_tri ){
      triangle[neigh[id_tri][points_together[i]-1]][j+3] =
                            tr[points_together[j]+3-1];
      triangle[neigh[id_tri-1][points_together[i]-1]][j+6] = s_temp[i];
    }
  }
}
